crates/levm/src/transaction.rs

```diff
@@ -0,0 +1,70 @@
+use bytes::Bytes;
+use ethereum_types::{Address, H256, U256};
+
+type AccessList = Vec<(Address, Vec<U256>)>;
+// type VersionedHash = H32;
+
+/// Transaction destination.
+#[derive(Clone, Debug)]
+pub enum TransactTo {
+    /// Simple call to an address.
+    Call(Address),
+    /// Contract creation.
+    Create,
+}
+
+/// The transaction environment.
+#[derive(Clone, Debug)]
+pub struct TxEnv {
+    /// Caller aka Author aka transaction signer.
+    pub caller: Address,
+    /// The gas limit of the transaction.
+    pub gas_limit: u64,
+    /// The gas price of the transaction.
+    pub gas_price: U256,
```

Comment:
gas_price should be an Option as well, as it only exists on the types Legacy and AccessList

Reply:
It is nonetheless calculated at some point (effective gas price). I'll wrap it in a option for now, and then we will see how to adapt it.